saleor/graphql/product/bulk_mutations/product_bulk_create.py

```diff
@@ -0,0 +1,832 @@
+                        message=e.messages[0],
+                        code=code,
+                    )
+                )
+
+    @classmethod
+    def clean_attributes(cls, cleaned_input, product_index, index_error_map):
+        attributes_errors_count = 0
+
+        if attributes := cleaned_input.get("attributes"):
+            try:
+                attributes_qs = cleaned_input["product_type"].product_attributes.all()
+                attributes = AttributeAssignmentMixin.clean_input(
+                    attributes, attributes_qs
+                )
+                cleaned_input["attributes"] = attributes
+            except ValidationError as exc:
+                if hasattr(exc, "error_dict"):
+                    cls.add_indexes_to_errors(
+                        product_index, exc, index_error_map, "attributes"
+                    )
+                else:
+                    index_error_map[product_index].append(
+                        ProductBulkCreateError(
+                            path="attributes",
+                            message=exc.message,
+                            code=exc.code,
+                        )
+                    )
+                attributes_errors_count += 1
+        return attributes_errors_count
+
+    @classmethod
+    def _clean_channel_listing(
+        cls, listings_data, channels_global_ids, product_index, index_error_map
+    ):
+        input_channel_ids = [
+            channel_listing["channel_id"] for channel_listing in listings_data
+        ]
+
+        wrong_channel_ids = {
+            channel_id
+            for channel_id in input_channel_ids
+            if channel_id not in channels_global_ids.keys()
+        }
+        if wrong_channel_ids:
+            index_error_map[product_index].append(
+                ProductBulkCreateError(
+                    path="channelListings",
+                    message="Not existing channel ID.",
+                    code=ProductBulkCreateErrorCode.NOT_FOUND.value,
+                    channels=wrong_channel_ids,
+                )
+            )
+
+        duplicates = get_duplicated_values(input_channel_ids)
+        if duplicates:
+            index_error_map[product_index].append(
+                ProductBulkCreateError(
+                    path="channelListings",
+                    message="Duplicated channel ID.",
+                    code=ProductBulkCreateErrorCode.DUPLICATED_INPUT_ITEM.value,
+                    channels=duplicates,
+                )
+            )
+        return duplicates, wrong_channel_ids
+
+    @staticmethod
+    def set_available_for_purchase_at(
+        is_available_for_purchase, available_for_purchase_at, channel_data
+    ):
+        if is_available_for_purchase is False:
+            channel_data["available_for_purchase_at"] = None
+        elif is_available_for_purchase is True and not available_for_purchase_at:
+            channel_data["available_for_purchase_at"] = datetime.now(pytz.UTC)
```

Comment:
Do we have the same logic for productCreate mutation? Look like a logic that should be defined in one place and possibly reused.

Reply:
Its not used in `productCreate` because in single mutation we are not able to add channel listings